src/inmanta/parser/plyInmantaParser.py

```diff
@@ -709,23 +703,29 @@ def p_constant_f(p: YaccProduction) -> None:
     attach_lnr(p)
 
 
-format_regex = r"""({{\s*([\.A-Za-z0-9_-]+)\s*}})"""
-format_regex_compiled = re.compile(format_regex, re.MULTILINE | re.DOTALL)
+def p_constant_string(p: YaccProduction) -> None:
+    " constant : STRING "
+    p[0] = get_string_ast_node(p[1], Location(file, p.lineno(1)))
+    attach_lnr(p)
+
+
+def p_constant_mls(p: YaccProduction) -> None:
+    " constant : mls "
+    p[0] = get_string_ast_node(p[1], p[1].location)
+    attach_from_string(p)
 
 
-def p_string(p: YaccProduction) -> None:
-    " constant : STRING "
-    value = p[1]
-    match_obj = format_regex_compiled.findall(str(value))
+def get_string_ast_node(string: LocatableString, location: Location) -> Union[Literal, StringFormat]:
+    format_regex = r"""({{\s*([\.A-Za-z0-9_-]+)\s*}})"""
+    format_regex_compiled = re.compile(format_regex, re.MULTILINE | re.DOTALL)
```

Comment:
pre-compile the regex for performance

Reply:
Of course. I don't know how I missed that.